### ISSUE TITLE
A question in running log

### ISSUE THREAD

**Issue by dufan175** [sunlicai/MAE-DFER#11]
I'm sorry to bother you again and again. Based on your work, I made some fine-tuning on my private dataset and the results were very good. However, there is a question in the running logs that I don't understand. As shown below, number of the training set is 1195 and number of the test set is 235. I want to know how **940** is obtained in "_Accuracy of the network on the  **940** test videos using last epoch model_", and I noticed that this number is usually four times the number of test sets. Is there any relationship between them? I have reviewed the code, but I still cannot understand it. Looking forward to your reply. Thank you very much again.
Here is the log:

Computing final results
235
Accuracy of the network on the  940 test videos using last epoch model: Top-1: 88.09%, Top-5: 100.00%
Total test samples: 235
Confusion Matrix:
[[ 90  20]
 [  8 117]]
Class Accuracies: ['81.82%', '93.60%']
UAR: 87.71%, WAR: 88.09%
Weighted F1: 0.8801, micro F1: 0.8809, macro F1: 0.8793
Training time 0:00:13

**Comment by youcaiSUN** [comment]
Hi @dufan175 , good question! 

During inference, MAE-DFER uniformly samples 2 clips along the *temporal* dimension by default (specified by the *test_num_segment* argument). And for each clip, it crop 2 *spatial* views (specified by the *test_num_crop* argument). Therefore, the final test video clips is actually 4 times larger than the original test set. 

https://github.com/sunlicai/MAE-DFER/blob/175521b5329bb30a31bc9b3de21724383db765b7/kinetics.py#L77-L84 

The 4 predictions of one test video are averaged to get the final prediction (details can be referred to *merge()* func in *engine_for_finetuning.py*). 

You can also find the actual number of test videos in your log: "Total test samples: 235".

**Comment by dufan175** [comment]
Oh, I understand. So if the test videos are too short to samples 2 clips, do i just need to set the _test_num_segment_ and  _test_num_crop_ to 1?

**Comment by youcaiSUN** [comment]
You don't need to do this. Actually, this part of code is from VideoMAE. Setting their values to 1 will cause running time errors. Besides, if the test video is too short, the code will sample two indentical clips. Thus, it won't influence the final prediction.

**Comment by dufan175** [comment]
Ok, I understand. Thank you for your reply.
